client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoResultSet.java

```diff
@@ -156,13 +158,13 @@ private static <T> Iterator<T> flatten(Iterator<Iterable<T>> iterator, long maxR
         // Semaphore to indicate that some data is ready.
         // Each permit represents a row of data (or that the underlying iterator is exhausted).
         private final Semaphore semaphore = new Semaphore(0);
-        private final CompletableFuture<Void> future;
+        private final Future<?> future;
 
         public AsyncIterator(Iterator<T> dataIterator, StatementClient client)
         {
             requireNonNull(dataIterator, "dataIterator is null");
             this.client = client;
-            this.future = CompletableFuture.runAsync(() -> {
```

Comment:
Looks good.
Now I wonder why we have used a CompletableFuture in the first place.